app/bundles/EmailBundle/Swiftmailer/Momentum/Metadata/MetadataProcessor.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+/*
+ * @copyright   2018 Mautic Contributors. All rights reserved
+ * @author      Mautic, Inc.
+ *
+ * @link        https://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\EmailBundle\Swiftmailer\Momentum\Metadata;
+
+use Mautic\EmailBundle\Helper\MailHelper;
+use Mautic\EmailBundle\Swiftmailer\Message\MauticMessage;
+
+class MetadataProcessor
+{
+    /**
+     * @var array
+     */
+    private $metadata = [];
+
+    /**
+     * @var array
+     */
+    private $substitutionKeys = [];
+
+    /**
+     * @var array
+     */
+    private $substitutionMergeVars = [];
+
+    /**
+     * @var array
+     */
+    private $mauticTokens = [];
+
+    /**
+     * @var \Swift_Message
+     */
+    private $message;
+
+    /**
+     * MetadataProcessor constructor.
+     *
+     * @param \Swift_Message $message
+     */
+    public function __construct(\Swift_Message $message)
+    {
+        $this->message = $message;
+
+        $metadata       = ($message instanceof MauticMessage) ? $message->getMetadata() : [];
+        $this->metadata = $metadata;
+
+        // Build the substitution merge vars
+        $this->buildSubstitutionData();
+
+        if (count($this->mauticTokens)) {
+            // Update the content with the substitution merge vars
+            MailHelper::searchReplaceTokens($this->mauticTokens, $this->substitutionMergeVars, $this->message);
+        }
+    }
+
+    /**
+     * @param $email
+     *
+     * @return array|mixed
+     */
+    public function getMetadata($email)
```

Comment:
There is a logic performed in the `getSubstitutionData` method. Why do we need another logic here, in a getter?